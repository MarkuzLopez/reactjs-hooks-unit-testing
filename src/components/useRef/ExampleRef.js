import React, { useState } from 'react';
import MultipleCustomHooks from '../example/MultipleCustomHooks';

const ExampleRef = () => {

    const [show, showState] = useState(false)

    const hideShow = () => {
        showState(!show)
    }

    return (  
        <>
            <h1>Ref</h1>
            <hr />
            { show && <MultipleCustomHooks /> }
            <br />
            <button className="btn btn-primary mt-5" onClick={hideShow}>Show/Hide</button>
        </>
    );
}
 
export default ExampleRef;