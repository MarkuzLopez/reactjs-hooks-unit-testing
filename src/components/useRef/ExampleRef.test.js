import { shallow } from "enzyme";
import ExampleRef from "./ExampleRef"

describe('Pruebas en <ExampleRef />', () => {

    const wrapper = shallow(<ExampleRef/>);

    test('debe de mostrarce de manera correctament', () => {
        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('MultipleCustomHooks').exists() ).toBe(false); // no va existir el componente cuando inicial la app

    });
    
    test('debe de mostrar el componente <MultipleCustomHooks /> ', () => {
        wrapper.find('button').simulate('click');
        expect(wrapper.find('MultipleCustomHooks').exists() ).toBe(true);
    })
    
})
