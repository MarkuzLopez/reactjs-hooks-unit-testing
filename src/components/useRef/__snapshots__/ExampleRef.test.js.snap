// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <ExampleRef /> debe de mostrarce de manera correctament 1`] = `
<Fragment>
  <h1>
    Ref
  </h1>
  <hr />
  <br />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</Fragment>
`;
